docs/framework/interop/default-marshaling-for-strings.md

```diff
@@ -1,264 +1,267 @@
+
+### Type Library Representation
+
+```cpp
+struct StringInfoA
+{
+    char *  f1;
+    char    f2[256];
+};
+
+struct StringInfoW
+{
+    WCHAR * f1;
+    WCHAR   f2[256];
+    BSTR    f3;
+};
+
+struct StringInfoT
+{
+    TCHAR * f1;
+    TCHAR   f2[256];
+};
+```
+
+The following code example shows how to use the <xref:System.Runtime.InteropServices.MarshalAsAttribute> attribute to define the same structure in different formats.
+
+```vb
+<StructLayout(LayoutKind.Sequential, CharSet := CharSet.Ansi)> _
+Structure StringInfoA
+    <MarshalAs(UnmanagedType.LPStr)> Public f1 As String
+    <MarshalAs(UnmanagedType.ByValTStr, SizeConst := 256)> _
+    Public f2 As String
+End Structure
+
+<StructLayout(LayoutKind.Sequential, CharSet := CharSet.Unicode)> _
+Structure StringInfoW
+    <MarshalAs(UnmanagedType.LPWStr)> Public f1 As String
+    <MarshalAs(UnmanagedType.ByValTStr, SizeConst := 256)> _
+    Public f2 As String
+<MarshalAs(UnmanagedType.BStr)> Public f3 As String
+End Structure
+
+<StructLayout(LayoutKind.Sequential, CharSet := CharSet.Auto)> _
+Structure StringInfoT
+    <MarshalAs(UnmanagedType.LPTStr)> Public f1 As String
+    <MarshalAs(UnmanagedType.ByValTStr, SizeConst := 256)> _
+    Public f2 As String
+End Structure
+```
+
+```csharp
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+struct StringInfoA
+{
+    [MarshalAs(UnmanagedType.LPStr)] public string f1;
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 256)] public string f2;
+}
+
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+struct StringInfoW
+{
+    [MarshalAs(UnmanagedType.LPWStr)] public string f1;
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 256)] public string f2;
+    [MarshalAs(UnmanagedType.BStr)] public string f3;
+}
+
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
+struct StringInfoT
+{
+    [MarshalAs(UnmanagedType.LPTStr)] public string f1;
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 256)] public string f2;
+}
+```
+
+## Fixed-Length String Buffers
+
+In some circumstances, a fixed-length character buffer must be passed into unmanaged code to be manipulated. Simply passing a string does not work in this case because the callee cannot modify the contents of the passed buffer. Even if the string is passed by reference, there is no way to initialize the buffer to a given size.
+
+The solution is to pass a <xref:System.Text.StringBuilder> buffer as the argument instead of a string. A `StringBuilder` can be dereferenced and modified by the callee, provided it does not exceed the capacity of the `StringBuilder`. It can also be initialized to a fixed length. For example, if you initialize a `StringBuilder` buffer to a capacity of `N`, the marshaler provides a buffer of size (`N`+1) characters. The +1 accounts for the fact that the unmanaged string has a null terminator while `StringBuilder` does not.
```

Comment:
```suggestion
The solution is to pass a <xref:System.Text.StringBuilder> buffer as the argument instead of a <xref:System.String>. A `StringBuilder` can be dereferenced and modified by the callee, provided it does not exceed the capacity of the `StringBuilder`. It can also be initialized to a fixed length. For example, if you initialize a `StringBuilder` buffer to a capacity of `N`, the marshaler provides a buffer of size (`N`+1) characters. The +1 accounts for the fact that the unmanaged string has a null terminator while `StringBuilder` does not.
```